jitsi-media-transform/src/main/kotlin/org/jitsi/nlj/util/ArrayCache.kt

```diff
@@ -178,27 +181,27 @@ open class ArrayCache<T>(
             return false
         }
 
-        val position = (head + diff) floorMod size
+        val position = position(diff)
         return (cache[position].index == index)
     }
 
     /**
      * Updates the [timeAdded] value of an item with a particular index, if it is in the cache.
      */
-    protected fun updateTimeAdded(index: Int, timeAdded: Long) = if (synchronize) {
+    protected fun updateTimeAdded(index: Long, timeAdded: Long) = if (synchronize) {
         synchronized(syncRoot) {
             doUpdateTimeAdded(index, timeAdded)
         }
     } else {
         doUpdateTimeAdded(index, timeAdded)
     }
 
-    private fun doUpdateTimeAdded(index: Int, timeAdded: Long) {
+    private fun doUpdateTimeAdded(index: Long, timeAdded: Long) {
         if (head == -1 || index > cache[head].index) {
             return
         }
         val diff = cache[head].index - index
-        val position = (head - diff) floorMod size
+        val position = position(diff)
```

Comment:
`position(diff)` does `head + diff`, but the original code here does `head - diff`